### ISSUE TITLE
The request of a dependency is an expression [webpack]

### ISSUE THREAD

**Issue by moshfeu** [fluent-ffmpeg/node-fluent-ffmpeg#862]
Disclaimer: Maybe it's non issue.

I'm using this library (part of [youtube-mp3-downloader](https://github.com/ytb2mp3/youtube-mp3-downloader)) with `webpack@4.21.0` and I get the warning below.

As much as I understand, webpack doesn't "like" dynamically `require`. You can get more info [here](https://github.com/webpack/webpack/issues/196#issuecomment-222084788).

### Version information

* fluent-ffmpeg version: 2.1.2
* ffmpeg version: irelevant
* OS: macos

### Expected results

This library will not cause webpack to warn

### Observed results

Webpack does warn about:

> /node_modules/fluent-ffmpeg/lib/options/misc.js 27:21-40
Critical dependency: the request of a dependency isan expression

### Checklist

* [ ] ~~I tried the same with command line ffmpeg and it works correctly (hint: if the problem also happens this way, this is an ffmpeg problem and you're not reporting it to the right place)~~ - Irelevant
* [ ] ~~I have included full stderr/stdout output from ffmpeg~~ - Irelevant

Thanks!

**Comment by njoyard** [comment]
Yes, this is known and is a side effect of the way our coverage testing is implemented.

If anyone comes up with a better solution, I'd be glad to merge it :wink:

**Comment by umens** [comment]
hey @moshfeu i found a way to fix it. check the MR #1077 while waiting for it to be merged and (maybe :smiley: ) release

**Comment by jackple** [comment]
temporary solution with webpack externals:

```js
 const IGNORES = ['fluent-ffmpeg']
 return function (context, request, callback) {
        if (IGNORES.includes(request)) {
            return callback(null, `require('${request}')`)
        }
       return callback()
 }
```
